### DISCUSSION TITLE
Bot/spam contributors - annoying PRs, future security risk?

### DISCUSSION THREAD

**Discussion by aormsby** [community/community#40419] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

I manage a public repo where recently I've been receiving [a lot of junk PRs](https://github.com/aormsby/Fork-Sync-With-Upstream-action/pull/63/files) from the [same individual.](https://github.com/cfsxy) They're a mess - single-line files, wildly divergent histories, just silly things. But the title for the PR never changes, the user repeatedly comments 'update', and then eventually... they just close the PR. This can all happen within an hour.

At first, I thought it was some novice dev who sucks at making PRs (we've all been there). But I started looking at their repos, and I changed my mind. It looks like all their activity is forking repos and simply adding/deleting project files en masse, and none of it had any functional significance that I could detect offhand. **Are they bot accounts? At the moment, it sure seems that way.**

But I wondered - what does someone gain from this behavior? And I hit on something. These accounts are generating _**false credibility over time.**_ The way GitHub contributions are counted is _super easy_ to manipulate, and the result is a lovely green contribution graph that someone could use to show they are reliable, trustworthy developers - when in this case they wouldn't be.

I know what you might be thinking:
- It's easy to see their 'work' is fake. -> Not if they delete all those forked repos later and make it impossible to see what the contributions were. Still shows on the graph.
- Report the user for abuse. -> Well, it's not abusive behavior in how GitHub defines it. Not currently harmful or aggressive.
- Block that user and move on, dude! -> Yeah, I probably will. But that doesn't prevent whatever's going on, and something seems _off_.

But some nagging feeling tells me that attempting to generate false credibility for a steady 3 years and counting is a security issue in the making. I'm not a security expert, but this kind of looks like the same tactics that subversive groups have used to develop 'credible' accounts on platforms like Twitter and Facebook. Be around for years, show vague but positive activity, and then one day... go HAM on some crazy attack or campaign or whatever.

Do I know what I'm talking about? Meh, who knows?

But what do you think? Is this something to discuss?

And at the very least, should Github have a way to report this kind of thing outside of 'abuse' and long-winded community posts? 🙂

**Comment by ossanna16** [comment]
Hi there @aormsby and welcome to our community! Thank you for asking a great question 🙂
While you are waiting for a reply to your question, here are a few things you can do:

To get started, introduce yourself in our [official introduction thread](https://github.com/orgs/community/discussions/23204)
To connect with other new users, check out this Discussion to [Meet Your GitHub Community Team](https://github.com/orgs/community/discussions/28776), further your GitHub knowledge by working through our [GitHub Skill 🆙 exercises](https://github.com/orgs/community/discussions/32121), and participate in our [weekly Fun Friday chats](https://github.com/orgs/community/discussions/30912)
